#4818: Update string renderer to only escape undisplayable code points.

rkeen-siemens opened the pull request:
Text displayed in the property sheet is escaped if the target font does not have a glyph for some of the characters. The existing implementation escapes some Unicode characters even if the target font can display them. This is due to the reliance on `Font.canDisplay(char)` which does not handle all Unicode characters.

The solution here is to update `RendererFactory.makeDisplayable` to iterate over the code points in the string rather than the characters. It then uses `Font.canDisplay(int)` to validate the glyph availability using the code point.

Unit tests are included to verify the existing behavior is maintained. All passed for the original code except `testDoesNotEscapeDisplayableCodePoint`.

|Original|Updated|
|----|----|
|<img width="515" alt="Original (View)" src="https://user-images.githubusercontent.com/46196614/229578373-3ce79fb1-fd8a-4cf3-ada9-3a6c8a454ac0.png"><br>When viewing the 𝑣 character renders correctly in the design view but shows as \ud835\udc63 in the property sheet|<img width="499" alt="Updated (View)" src="https://user-images.githubusercontent.com/46196614/229578376-a000de90-30d8-472b-a420-1552b8078052.png"><br>The 𝑣 character is now rendered as expected but the undisplayable ⑙ character (at least on my system) still shows the unicode escape|
|<img width="496" alt="Updated (Edit)" src="https://user-images.githubusercontent.com/46196614/229576915-b0867a3c-9279-4424-8ddb-36b1547ca84b.png">|<img width="489" alt="Original (Edit)" src="https://user-images.githubusercontent.com/46196614/229576924-c4c8a46c-7cd0-4b0c-b3e9-afb0c875f8bc.png"><br>Edit mode is unchanged|

neilcsmith-net commented:
Thanks for the PR.

Locked and unlocked to trigger the right tests (it appears that the unapproved test run doesn't pick up the label addition).

mbien commented:
> it appears that the unapproved test run doesn't pick up the label addition at approval

@neilcsmith-net yes this is super annoying. I usually remove the not-approved workflow run from [here](https://github.com/apache/netbeans/actions) first, then start a new one with the lock trick. This is also mentioned as last point of the "important" section on the [wiki](https://cwiki.apache.org/confluence/display/NETBEANS/PRs+and+You+-+A+reviewer+Guide).

mbien commented:
@rkeen-siemens could you add a screenshot to the PR? This would be linked from the release notes, screenshots are often helpful for visual changes.

mbien reviewed (approved):
code looks good to me

however, before merge please update the email address on the commit, so that it doesn't ues a noreply one, feel free to force push.

you can check it via:
https://github.com/apache/netbeans/pull/5757.patch

rkeen-siemens commented:
> could you add a screenshot to the PR

Added before and after views for edit and view mode

> please update the email address on the commit

Updated

mbien commented:
@rkeen-siemens thanks again for the contribution, all green again -> merging.
